### ISSUE TITLE
Add moth/amq1.py (AMQP 1.0 support using qpid.)

### ISSUE THREAD

**Issue by petersilva** [MetPX/sarracenia#389]
Support of this message protocol is necessary, but not sufficient, for aviation clients.  
It is necessary, because aviation standard SWIM, has supposedly standardized on use of AMQP 1.0, without specifying the payload.  It seems they do not support third party transfers however, so all messages must be inlined, and the inlining is different from what we do in WMO, so will set a separate issue for swim support, after amqp support (should be a derived class from amq1)

**Comment by reidsunderland** [comment]
This has become a reasonably high priority now. @habilinour hopes to have some progress by the end of July (a proof of concept).

@andreleblanc11 and I started testing RabbitMQ 4.1 with built in AMQP 1.0 support, trying to shovel AMQP 1.0 to AMQP 0.9.1 but didn't get very far. The messages ended up in a format we couldn't decode and it seemed like it would be a better use of time to start working on AMQP 1.0 support in sr3 instead.

We found some documentation and a demonstration broker that we can connect to:
- https://swim.iblsoft.com/swimdemo/latest/SWIM-Weather-Public-Demonstration/
- Documentation of the MET-SWIM message format: https://github.com/iblsoft/swimdemo/blob/main/MET-SWIM-AMQP-Guidance.md#amqp-message-structure-overview
- Example code: https://github.com/iblsoft/swimdemo/blob/main/amqp_client_example.py

I've started working on sr3 code on branch [issue389](https://github.com/MetPX/sarracenia/tree/issue389). I can connect to the broker, receive messages, download the data when messages include links. But there's still a lot of work ahead.

I haven't made any attempt to publish messages yet, just ingest.

@junhu3

**Comment by petersilva** [comment]
It's great to see that you found the postformat class hierarchy... that's prep work that was done to make adding swim much easier to add than it used to be.

**Comment by reidsunderland** [comment]
Yeah, that's working great. It would have been way more work if the postformat and moth classes weren't already set up. :)
